appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java

```diff
@@ -1031,11 +1031,17 @@ private boolean insertObjectReference(ApiContext context, Reference<?> referee,
             return false;
         }
 
+        final Schema schema = AnnotationInfo.valueOf(referenceClass).getAnnotation(Schema.class);
+
         // Set the reference name
-        referee.setRef(referenceClass.getSimpleName());
+        if (schema == null || schema.name().isEmpty()) {
+            referee.setRef(referenceClass.getSimpleName());
+        } else {
+            referee.setRef(schema.name());
+        }
```

Comment:
I'd prefer a single `referee.setRef(...)` with a conditional expression but that is personal preference.

Reply:
Why not ? Can be squeezed into a single line quite easily. 

```
referee.setRef(schema == null || schema.name().isEmpty() ? referenceClass.getSimpleName() : schema.name());
```

Changed in https://github.com/payara/Payara/pull/4494/commits/2e2e6f6f7a85351e4243eaf6fbf2b6b9248a5bc2.

Reply:
Don't know why, some people do not like conditionals.

Reply:
ternary operators make it far less readable, compiler can do this optimization.

Reply:
> ternary operators make it far less readable

That is subjective. I can read conditions "better".

> compiler can do this optimization.

Its not a readability or performance optimization as far as I am concerned but a semantic one.  With the conditional it is very clear that we want to do this (here `setRef`) with some value in any branch. This is much clearer expressed by not doing that in an `if-else` where both branches do it but a single call with a argument that is either this or that.

Reply:
I'm fine with both. I can read almost any code of any style. Feel free to adjust it to your liking.